### ISSUE TITLE
GHI Auth

### ISSUE THREAD

**Issue by narthur** [benjlevesque/circleci-orbs#1]
First time using GHI or orbs. How do you go about generating your auth token and making it accessible to the orb?

**Comment by narthur** [comment]
I've made some progress:

- Installed GHI on my local machine: `brew install ghi`
- Generated an auth token: `ghi config --auth [username]`
- Copied the token from Keychain
- Created a new GHI_TOKEN environment variable in CircleCI with the token as its value

The job now successfully runs and comments on a pull request, but it's not the currently open pull request, it's an old, closed pull request associated with the same branch (a `dev` branch that gets multiple pull requests opened against it over time).

So, I guess my new question is, how do I ensure that the PR_ID variable it uses contains the correct, newest-open PR ID?

**Comment by narthur** [comment]
I've now followed the instructions [on this page](https://support.circleci.com/hc/en-us/articles/360006357533--CIRCLE-PULL-REQUEST-is-missing-from-my-Pull-Request) to tell CircleCI to only build commits associated with pull requests. My hypothesis is that, before I created the environment variable, I hadn't created the new PR, so when I reran the build after creating the environment variable, CircleCI didn't refresh the PR variable to point to the new variable.

It remains to be seen whether this will actually fix the issue.

**Comment by benjlevesque** [comment]
Hey @narthur , sorry about the late reply, i was on vacation

yes, `GHI_TOKEN` is the way to go 👍 

indeed, it needs to run on a PR, and the CircleCI option is a good solution imo, did it fix your issue?

**Comment by narthur** [comment]
@benjlevesque No worries. :)

I ended up doing something totally different, since I ran into more issues following my last comment:

1. After telling CircleCI to stop running builds on commits not associated with PRs, CircleCI started refusing to run commits even if they were associated with an open PR.
2. CircleCI's environment variables were pointing to an old, closed PR associated with the same branch, even though there was a much newer, open PR in existence.

For those reasons, I gave up on posting back to the pull request, and instead used Surge.sh to host the latest build with a URL based on the current branch name, and then added some badges linking to those Surge urls to the project's readme.
